docs/feature_combo.md

```diff
@@ -67,31 +86,100 @@ void process_combo_event(uint8_t combo_index, bool pressed) {
+
+#### Permissive Hold
+Similar to Tap-Hold configuration, Mod Combos can be setup to behave similarly. Add `#define COMBO_PERMISSIVE_HOLD` to your `config.h` to enable. This feature is extremely useful for fast typists as you don't have to wait for the Mod Combo's combo term to finish.
+
+If `COMBO_TERM` and `COMBO_MOD_TERM` are too close to each other this settings becomes less useful as this only affects if a key is pressed between `COMBO_TERM` and `COMBO_MOD_TERM`. All key presses under `COMBO_TERM` are always considered for overlapping combos.
+
+#### Per Combo Timing
+Instead of using a blanket `COMBO_TERM` window for every combo, per-combo timings can be generated. In order to use this feature the following config options and functions can be defined. Coming up with useful timings and configuration is left as an exercise for the reader.
+
+| Config Flag                 | Function                                                  | Description                                                                                          |
+|-----------------------------|-----------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `COMBO_MUST_HOLD_PER_COMBO` | bool get_combo_must_hold(uint16_t index, combo_t \*combo) | Controls if a given combo should fire immediately on tap or if it needs to be held. (default: False) |
+| `COMBO_TERM_PER_COMBO`      | uint16_t get_combo_term(uint16_t index, combo_t \*combo)  | Optional per-combo timeout window. (default: COMBO_TERM)                                             |
+
+If you set either of the options but like the Modifier Combo settings, you have to define those yourself in the function.
+
+Examples:
+```c
+uint16_t get_combo_term(uint16_t index, combo_t *combo) {
+    // decide by combo->keycode
+    if (IS_MOD(combo->keycode)) return COMBO_MOD_TERM; // you have to config this yourself ifdef COMBO_TERM_PER_COMBO
+
+    switch (combo->keycode) {
+        case KC_X:
+            return 50;
+    }
+
+    // or with combo index, i.e. its name from enum.
+    switch (index) {
+        case COMBO_NAME_HERE:
+            return 9001;
+    }
+
+    return COMBO_TERM;
+}
+
+bool get_combo_must_hold(uint16_t index, combo_t *combo) {
+    // yet again, with keycode and true for Modifiers if so desired.
+    if (IS_MOD(combo->keycode)) return true;
+
+    // or index/name
+    switch (index) {
+        case COMBO_NAME_HERE:
+            return true;
+    }
+
+    return false;
+}
+```
+
+#### Variable Length Combos
+If `COMBO_VARIABLE_LEN` is defined allows the user to pragmatically declare the size of the Combo data structure and avoid updating COMBO_COUNT. Instead a variable is exported called COMBO_LEN and is usually set with something similar to the following in `keymap.c`: `int COMBO_LEN = sizeof(key_combos) / sizeof(key_combos[0]);`
```

Comment:
« If `COMBO_VARIABLE_LEN` is defined allows the user to » → « If `COMBO_VARIABLE_LEN` is defined, it allows the user to ».

Is « pragmatically » intended or is it actually supposed to say « to **programmatically** declare »?